multi/imxrt-multi/cm4.c

```diff
@@ -169,68 +183,75 @@ static int mu_irqHandler(unsigned int n, void *arg)
 
 static ssize_t chanRead(id_t id, void *buff, size_t bufflen)
 {
-	char *tbuff = buff;
-	ssize_t i;
+	unsigned char *tbuff = buff;
+	size_t i;
 	fifo_t *fifo;
 
-	if (id < chan0 || id > chan3 || buff == NULL)
-		return -1;
+	if ((id < chan0) || (id > chan3) || (buff == NULL)) {
+		return -EINVAL;
+	}
 
 	fifo = &m4_common.channel[id].rx;
 
 	for (i = 0; i < bufflen; ++i) {
-		if (!fifo_pop(&tbuff[i], fifo))
+		if (fifo_pop(&tbuff[i], fifo) == 0) {
 			break;
+		}
 	}
 
-	return i;
+	return (ssize_t)i;
 }
 
 
 static ssize_t chanWrite(id_t id, const void *buff, size_t bufflen)
 {
-	const char *tbuff = buff;
-	int i;
+	const unsigned char *tbuff = buff;
+	size_t i;
 	fifo_t *fifo;
 
-	if (id < chan0 || id > chan3 || buff == NULL)
-		return -1;
+	if ((id < chan0) || (id > chan3) || (buff == NULL)) {
+		return -EINVAL;
+	}
 
 	fifo = &m4_common.channel[id].tx;
 
 	for (i = 0; i < bufflen; ++i) {
-		if (!fifo_push(tbuff[i], fifo))
+		if (fifo_push(tbuff[i], fifo) == 0) {
 			break;
+		}
 	}
 
-	if (i)
+	if (i != 0) {
 		setTXirq(id, 1);
+	}
 
-	return i;
+	return (ssize_t)i;
 }
 
 
 static int runCore(unsigned int offset)
 {
 	platformctl_t pctl;
-	unsigned int vectors = (unsigned int)M4_MEMORY_START + offset;
+	unsigned int vectors = (unsigned int)CM4_MEMORY_START + offset;
 
 	/* Set CM4's VTOR to the start of it's memory */
 	pctl.action = pctl_set;
 	pctl.type = pctl_iolpsrgpr;
 	pctl.iogpr.field = 0;
 	pctl.iogpr.val = vectors & 0xffff;
```

Comment:
btw. may be corrected, this should be:
```suggestion
	pctl.iogpr.val = vectors & 0xfff8;
```
LPSR_GPR0 bits 2-0 of CM4_INIT_VTOR_LOW are reserved.